config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/config/loader/PolicyAndFolderLoader.java

```diff
@@ -75,6 +82,23 @@ private void loadPolicies(final File currentDir, final File rootDir, Folder pare
         }
     }
 
+    private Map<Dependency, List<Dependency>> getPolicyDependencies(final File policyRootDir) {
+        final Map<Dependency, List<Dependency>> policyDependencyMap = new HashMap<>();
+        File policyMetadataFile = new File(policyRootDir, "policies" + jsonTools.getFileExtension());
+        final ObjectMapper objectMapper = jsonTools.getObjectMapper();
+        final MapType type = objectMapper.getTypeFactory().constructMapType(HashMap.class, String.class, PolicyMetadata.class);
+        final Map<String, PolicyMetadata> policyMetadataMap = jsonTools.readDocumentFile(policyMetadataFile, type);
+        if (policyMetadataMap != null) {
+            for(Map.Entry<String, PolicyMetadata> metadataEntry : policyMetadataMap.entrySet()){
+                final String policyNameWithPath = metadataEntry.getKey();
+                int index = policyNameWithPath.lastIndexOf("/");
+                final String policyName = index > -1 ? policyNameWithPath.substring(index+1) : policyNameWithPath;
```

Comment:
Can't you get the policy name alone from the policy-metadata?

Reply:
we are not saving name in the policies.yml